src/dataframerow/utils.jl

```diff
@@ -94,6 +94,68 @@ isequal_row(cols1::Tuple{Vararg{AbstractVector}}, r1::Int,
     isequal(cols1[1][r1], cols2[1][r2]) &&
         isequal_row(Base.tail(cols1), r1, Base.tail(cols2), r2)
 
+# Simple vector type for internal use which represents a virtual DataAPI.refpool
+# so that a vector of reals can be its own DataAPI.refarray:
+# this just allows telling the generic row_group_slots method for generic refpools
+# what is the minimum index and the number of (potential) groups
+# missing values are represented using index max+1
+struct IntegerRefpool{T<:Union{Int, Missing}} <: AbstractVector{T}
+    min::Int
+    max::Int
+    function IntegerRefpool{T}(min::Real, max::Real) where T<:Union{Int, Missing}
+        @assert max < typemax(Int) - 1
+        @assert typemin(Int) <= widen(max) - widen(min) + 1 < typemax(Int)
+        new{T}(min, max)
+    end
+end
+
+Base.size(x::IntegerRefpool{T}) where {T} = (x.max - x.min + 1 + (T >: Missing),)
+Base.axes(x::IntegerRefpool{T}) where {T} = (x.min:(x.max + (T >: Missing)),)
+Base.IndexStyle(::Type{<:IntegerRefpool}) = Base.IndexLinear()
+@inline function Base.getindex(x::IntegerRefpool{T}, i::Real) where T
+    @boundscheck checkbounds(x, i)
+    if T >: Missing && i == x.max + 1
+        return missing
+    else
+        return Int(i - x.min + 1)
+    end
+end
+Base.allunique(::IntegerRefpool) = true
+Base.issorted(::IntegerRefpool) = true
+
+function refpool_and_array(x::AbstractArray)
+    refpool = DataAPI.refpool(x)
+    refarray = DataAPI.refarray(x)
+
+    if refpool !== nothing
+        return refpool, refarray
+    elseif x isa AbstractArray{<:Union{Real, Missing}} &&
+        all(v -> ismissing(v) | isinteger(v), x)
+        isempty(skipmissing(x)) && return nothing, nothing
+        minval, maxval = extrema(skipmissing(x))
+        # Threshold chosen with the same rationale as the row_group_slots refpool method:
+        # refpool approach is faster but we should not allocate too much memory either
+        # We also have to avoid overflow
+        if typemin(Int) <= maxval + 1 < typemax(Int) &&
+            typemin(Int) <= minval <= typemax(Int) &&
+            widen(maxval) - widen(minval) + 1 <= 2 * length(x) < typemax(Int)
```

Comment:
as above you have to use `+2` to allow to store `missing`
```suggestion
            widen(maxval) - widen(minval) + 2 <= 2 * length(x) < typemax(Int)
```